django_tenants/test/client.py

```diff
@@ -14,14 +14,16 @@ def __init__(self, tenant, **defaults):
     def generic(self, *args, **kwargs):
         if "HTTP_HOST" not in kwargs:
             kwargs["HTTP_HOST"] = self.tenant.get_primary_domain().domain
-        request = super().generic(*args, **kwargs)
-        # Assign the tenant to the request object
-        request.tenant = self.tenant
-        return request
+        return super().generic(*args, **kwargs)
 
 
 class TenantRequestFactory(BaseTenantRequestFactory, RequestFactory):
-    pass
+
+    def generic(self, *args, **kwargs):
+        request = super().generic(*args, **kwargs)
+        # Process request using Tenant middleware
+        self.tm.process_request(request)
+        return request
```

Comment:
I think this could benefit from explicit documentation inside README and/or doc-strings for the `BaseTenantRequestFactory` class.

In fact this library now supports several different middleware implementations (TenantMainMiddleware, TenantSubfolderMiddleware, SuspiciousTenantMiddleware) and folks may contribute more over time. 

Yet at present TenantClient defaults to a specific middleware implementation. That should be easier to override in one's tests.

Reply:
Seeing how we also have SubfolderTenantTestCase which doesn't appear to override the middleware (from what I can see adjusting the request client is left to the developer - that's what I have in my own tests). 

This has the potential to break lots of things in unexpected ways.